src/BlockEntities/ChestEntity.cpp

```diff
@@ -38,7 +40,11 @@ void cChestEntity::SendTo(cClientHandle & a_Client)
 	// The chest entity doesn't need anything sent to the client when it's created / gets in the viewdistance
 	// All the actual handling is in the cWindow UI code that gets called when the chest is rclked
 
-	UNUSED(a_Client);
+	// Temporary fix for invisible chests (#3479, #3403)
+	if (a_Client.GetProtocolVersion() >= cProtocolRecognizer::PROTO_VERSION_1_9_0) 
+	{
+		a_Client.SendUpdateBlockEntity(*this);
+	}
```

Comment:
According to the protocol wiki http://wiki.vg/Block_Actions#Chest , there is a chest block action, does sending that instead fix the problem? I'd prefer it as a cleaner solution.